cosmos/module.go

```diff
@@ -27,12 +28,15 @@ type AppModuleBasic struct {
 // AppModule is a main element of an cosmos app.
 type AppModule struct {
 	AppModuleBasic
-	handler sdk.Handler
+	handler Handler
 	querier Querier
 }
 
+// Handler defines the core of the state transition function of an application.
+type Handler func(request cosmostypes.Request, msg cosmostypes.Msg) (hash.Hash, cosmostypes.Error)
```

Comment:
Why forcing the type of the error?
There is no need to force the type. The handler function can just wrap the error to an cosmos error if needed. See https://github.com/mesg-foundation/engine/pull/1463/files/a54e6902fede8ef8ec23c70df1791243581d9c4f#diff-472d4db73a8fc27ef1b87e0e6c294afcR109-R114

```suggestion
type Handler func(request cosmostypes.Request, msg cosmostypes.Msg) (hash.Hash, error)
```

Reply:
That's true, to change.